src/main/java/it/pagopa/transactions/utils/AuthRequestDataUtils.java

```diff
@@ -0,0 +1,38 @@
+package it.pagopa.transactions.utils;
+
+import it.pagopa.ecommerce.commons.annotations.ValueObject;
+import it.pagopa.generated.transactions.server.model.OutcomeVposGatewayDto;
+import it.pagopa.generated.transactions.server.model.OutcomeXpayGatewayDto;
+import it.pagopa.generated.transactions.server.model.UpdateAuthorizationRequestDto;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class AuthRequestDataUtils {
+
+    @ValueObject
+    public record AuthRequestData(
+            String authorizationCode,
+            String outcome,
+            String rrn
+    ) {
+
+    }
+
+    public AuthRequestData extract(UpdateAuthorizationRequestDto updateAuthorizationRequest) {
+        AuthRequestData result = null;
+        switch (updateAuthorizationRequest.getOutcomeGateway()) {
+            case OutcomeVposGatewayDto t -> {
+                result = new AuthRequestData(t.getAuthorizationCode(),t.getOutcome().toString(),t.getRrn());
+            }
+            case OutcomeXpayGatewayDto t -> {
+                result = new AuthRequestData(t.getAuthorizationCode(),t.getOutcome().toString(), null);
+            }
+            default ->
+                    throw new IllegalStateException("Unexpected value: " + updateAuthorizationRequest.getOutcomeGateway());
```

Comment:
What about use here `it.pagopa.transactions.exceptions.InvalidRequestException` so that we will return a 400 bad request response (see controller exception handler related section)?